courier/template/load_template.go

```diff
@@ -61,12 +71,45 @@ func loadBuiltInTemplate(filesytem fs.FS, name string, html bool) (Template, err
 		tpl = t
 	}
 
-	_ = cache.Add(name, tpl)
+	_ = Cache.Add(name, tpl)
 	return tpl, nil
 }
 
+func loadRemoteTemplate(ctx context.Context, d templateDependencies, url string, html bool) (Template, error) {
+	var bb *bytes.Buffer
+	var err error
+
+	// instead of creating a new request always we always cache the bytes.Buffer using the url as the key
+	if t, found := Cache.Get(url); found {
+		bb = t.(*bytes.Buffer)
```

Comment:
Caching buffers is pretty difficultl. In HTTP for example, the buffer can only be read once because the server sends the body only once. Please use `[]byte` for caching instead

Reply:
Ah okay, makes sense!